### PR TITLE
Accept signedUpAt override in js widget

### PR DESCRIPTION
**Pull request by alavers**
@lemieux @mspensieri @dannytranlx @Mario54

### CODE REVIEW

**Comment by lemieux** [comment]
:+1:

**Comment by mspensieri** [comment]
While you're at it can you change line 228 in main.js

```
user.save(userInfo, {

// TO 

user.save(_.pick(userInfo, AppUser.EDITABLE_PROPERTIES), {
```

**Comment by mspensieri** [comment]
:+1:

File: src/js/models/appUser.js [hunk 1]
```diff
@@ -13,7 +13,9 @@ var AppUser = module.exports = Backbone.AssociatedModel.extend({
     },
 
     parse: function(data) {
-        return data.appUser;
+        var appUser = data.appUser;
+        appUser.signedUpAt = new Date(data.appUser.signedUpAt);
```
Comment (Mario54) [inline src/js/models/appUser.js]: This is necessary because the `signedUpAt` is stored as a number in the backend?

**Comment by Mario54** [comment]
:+1:

File: src/js/models/appUser.js (hunk 1 above)
Comment (lemieux) [inline src/js/models/appUser.js]: yep

Comment (alavers) [inline src/js/models/appUser.js]: to be more specific, `Date` gets serialized to a ISO 8601 string on the wire, so unless we do this we get back a string in the backbone model when we sync. I want it to be a `Date` going in and a date coming back out again.

Comment (Mario54) [inline src/js/models/appUser.js]: So, this takes care of parsing the response from the server, but what takes care of serializing `signedUpAt` when we make a request to the server? Or does the server accept a ISO 8601 string?

**Pull request merged**